test/functional/logforwarding/message_format_test.go

```diff
@@ -22,48 +38,100 @@ var _ = Describe("[LogForwarding] Functional tests for message format", func() {
 		framework = functional.NewFluentdFunctionalFramework()
 		functional.NewClusterLogForwarderBuilder(framework.Forwarder).
 			FromInput(logging.InputNameApplication).
+			ToFluentForwardOutput().
+			FromInput(logging.InputNameAudit).
 			ToFluentForwardOutput()
 		Expect(framework.Deploy()).To(BeNil())
 	})
 	AfterEach(func() {
 		framework.Cleanup()
 	})
 
+	It("should parse k8s audit log format correctly", func() {
+		// Log message data
+		timestamp := "2013-03-28T14:36:03.243000+00:00"
+		nanoTime, _ := time.Parse(time.RFC3339Nano, timestamp)
+		kind := fmt.Sprintf("audit(%.3f:24287)", float64(nanoTime.UnixNano())/float64(time.Second))
+
+		// Define a template for test format (used for input, and expected output)
+		var outputLogTemplate = types.K8sAuditLog{
+			AuditLogCommon: types.AuditLogCommon{
+				Kind:             kind,
+				ViaqIndexName:    "audit-write",
+				Level:            "info",
+				Timestamp:        time.Time{},
+				ViaqMsgID:        "*",
+				PipelineMetadata: templateForAnyCollector,
+			},
+		}
+		// Template expected as output Log
+		k8sAuditLogLine := "{\\\"kind\\\":\\\"" + kind + "\\\"}"
+		Expect(framework.WriteMessagesTok8sAuditLog(k8sAuditLogLine, 10)).To(BeNil())
+		// Read line from Log Forward output
+		raw, err := framework.ReadAuditLogsFrom("fluentforward")
+		Expect(err).To(BeNil(), "Expected no errors reading the logs")
+		var logs []types.K8sAuditLog
+		err = types.StrictlyParseLogs(raw, &logs)
+		Expect(err).To(BeNil(), "Expected no errors parsing the logs")
+		// Compare to expected template
+		outputTestLog := logs[0]
+		Expect(outputTestLog).To(FitLogFormatTemplate(outputLogTemplate))
+	})
+
+	It("should parse linux audit log format correctly", func() {
+		// Log message data
+		msgType := "CWD"
+		timestamp := "2013-03-28T14:36:03.243000+00:00"
+		nanoTime, _ := time.Parse(time.RFC3339Nano, timestamp)
+		msg := fmt.Sprintf("audit(%.3f:24287)", float64(nanoTime.UnixNano())/float64(time.Second))
+		auditLogLine := fmt.Sprintf("type=%s msg=%s:", msgType, msg)
+		// Template expected as output Log
+		var outputLogTemplate = types.LinuxAuditLog{
+			Message:       auditLogLine,
+			ViaqIndexName: "audit-write",
+			AuditLinux: types.AuditLinux{
+				Type:     msgType,
+				RecordID: "*"},
+			Timestamp:        nanoTime,
```

Comment:
This is a nit but something I was thinking about when you mentioned it.  We do desire the time to be the correct format but it doesn't matter if it is "time"; it should be a string.  JSON only supports the basic types anyway.  This may be but a detail.